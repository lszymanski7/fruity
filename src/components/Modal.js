import PropTypes from 'prop-types'
import React from 'react'
import ReactModal from 'react-modal'

const Modal = (props) => {
    const { selectedFruit, isOpenModal, closeModal } = props

    return (
        <ReactModal
            appElement={document.getElementById('root')}
            className="modal"
            contentLabel="Fruity Modal"
            isOpen={isOpenModal}
            onRequestClose={closeModal}
        >
            {
                selectedFruit && 
                <div>
                    <h2 className="modal__title">{selectedFruit.name}</h2>
                    <img className="modal__image" id={selectedFruit.id} src={selectedFruit.img.color} alt={selectedFruit.name} />
                </div>
            }
            <button className="button" onClick={closeModal}>Okay</button>
        </ReactModal>
    )
}

Modal.propTypes = {
	selectedFruit: PropTypes.object,
	isOpenModal: PropTypes.bool.isRequired,
    closeModal: PropTypes.func.isRequired
}

export { Modal as default }
