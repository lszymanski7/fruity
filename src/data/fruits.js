import { ICONS } from './constants'

// Fruits data
const fruits = [
    {
        id: 1,
        name: 'apple',
        img: ICONS.apple,
        checked: false
    },
    {
        id: 2,
        name: 'banana',
        img: ICONS.banana,
        checked: false
    },
    {
        id: 3,
        name: 'blackberry',
        img: ICONS.blackberry,
        checked: false
    },
    {
        id: 4,
        name: 'blueberries',
        img: ICONS.blueberries,
        checked: false
    },
    {
        id: 5,
        name: 'cherries',
        img: ICONS.cherries,
        checked: false
    },
    {
        id: 6,
        name: 'coconut',
        img: ICONS.coconut,
        checked: false
    },
    {
        id: 7,
        name: 'dragon fruit',
        img: ICONS.dragon_fruit,
        checked: false
    },
    {
        id: 8,
        name: 'fig',
        img: ICONS.fig,
        checked: false
    },
    {
        id: 9,
        name: 'grapes',
        img: ICONS.grapes,
        checked: false
    },
    {
        id: 10,
        name: 'kiwi',
        img: ICONS.kiwi,
        checked: false
    },
    {
        id: 11,
        name: 'lemon',
        img: ICONS.lemon,
        checked: false
    },
    {
        id: 12,
        name: 'lychee',
        img: ICONS.lychee,
        checked: false
    },
    {
        id: 13,
        name: 'mango',
        img: ICONS.mango,
        checked: false
    },
    {
        id: 14,
        name: 'melon',
        img: ICONS.melon,
        checked: false
    },
    {
        id: 15,
        name: 'orange',
        img: ICONS.orange,
        checked: false
    },
    {
        id: 16,
        name: 'papaya',
        img: ICONS.papaya,
        checked: false
    },
    {
        id: 17,
        name: 'peach',
        img: ICONS.peach,
        checked: false
    },
    {
        id: 18,
        name: 'pear',
        img: ICONS.pear,
        checked: false
    },
    {
        id: 19,
        name: 'pineapple',
        img: ICONS.pineapple,
        checked: false
    },
    {
        id: 20,
        name: 'plum',
        img: ICONS.plum,
        checked: false
    },
    {
        id: 21,
        name: 'pomegranate',
        img: ICONS.pomegranate,
        checked: false
    },
    {
        id: 22,
        name: 'star fruit',
        img: ICONS.star_fruit,
        checked: false
    },
    {
        id: 23,
        name: 'strawberry',
        img: ICONS.strawberry,
        checked: false
    },
    {
        id: 24,
        name: 'watermelon',
        img: ICONS.watermelon,
        checked: false
    }
]

export { fruits as default }
