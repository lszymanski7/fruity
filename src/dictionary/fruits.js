import { fruitsColor, fruitsLinear } from '../assets/icons/index'

const fruits = [{
    name: 'Apple',
    img: [
        fruitsColor['apple.png'].default,
        fruitsLinear['apple.png'].default
    ]
}, {
    name: 'Banana',
    img: [
        fruitsColor['banana.png'].default,
        fruitsLinear['banana.png'].default
    ]
}, {
    name: 'Blackberry',
    img: [
        fruitsColor['blackberry.png'].default,
        fruitsLinear['blackberry.png'].default
    ]
}, {
    name: 'Blueberries',
    img: [
        fruitsColor['blueberries.png'].default,
        fruitsLinear['blueberries.png'].default
    ]
}, {
    name: 'Cherries',
    img: [
        fruitsColor['cherries.png'].default,
        fruitsLinear['cherries.png'].default
    ]
}, {
    name: 'Coconut',
    img: [
        fruitsColor['coconut.png'].default,
        fruitsLinear['coconut.png'].default
    ]
}, {
    name: 'Dragon Fruit',
    img: [
        fruitsColor['dragon-fruit.png'].default,
        fruitsLinear['dragon-fruit.png'].default
    ]
}, {
    name: 'Fig',
    img: [
        fruitsColor['fig.png'].default,
        fruitsLinear['fig.png'].default
    ]
}, {
    name: 'Grapes',
    img: [
        fruitsColor['grapes.png'].default,
        fruitsLinear['grapes.png'].default
    ]
}, {
    name: 'Kiwi',
    img: [
        fruitsColor['kiwi.png'].default,
        fruitsLinear['kiwi.png'].default
    ]
}, {
    name: 'Lemon',
    img: [
        fruitsColor['lemon.png'].default,
        fruitsLinear['lemon.png'].default
    ]
}, {
    name: 'Lychee',
    img: [
        fruitsColor['lychee.png'].default,
        fruitsLinear['lychee.png'].default
    ]
}, {
    name: 'Mango',
    img: [
        fruitsColor['mango.png'].default,
        fruitsLinear['mango.png'].default
    ]
}, {
    name: 'Melon',
    img: [
        fruitsColor['melon.png'].default,
        fruitsLinear['melon.png'].default
    ]
}, {
    name: 'Orange',
    img: [
        fruitsColor['orange.png'].default,
        fruitsLinear['orange.png'].default
    ]
}, {
    name: 'Papaya',
    img: [
        fruitsColor['papaya.png'].default,
        fruitsLinear['papaya.png'].default
    ]
}, {
    name: 'Peach',
    img: [
        fruitsColor['peach.png'].default,
        fruitsLinear['peach.png'].default
    ]
}, {
    name: 'Pear',
    img: [
        fruitsColor['pear.png'].default,
        fruitsLinear['pear.png'].default
    ]
}, {
    name: 'Pineapple',
    img: [
        fruitsColor['pineapple.png'].default,
        fruitsLinear['pineapple.png'].default
    ]
}, {
    name: 'Plum',
    img: [
        fruitsColor['plum.png'].default,
        fruitsLinear['plum.png'].default
    ]
}, {
    name: 'Pomegranate',
    img: [
        fruitsColor['pomegranate.png'].default,
        fruitsLinear['pomegranate.png'].default
    ]
}, {
    name: 'Star Fruit',
    img: [
        fruitsColor['star-fruit.png'].default,
        fruitsLinear['star-fruit.png'].default
    ]
}, {
    name: 'Strawberry',
    img: [
        fruitsColor['strawberry.png'].default,
        fruitsLinear['strawberry.png'].default
    ]
}, {
    name: 'Watermelon',
    img: [
        fruitsColor['watermelon.png'].default,
        fruitsLinear['watermelon.png'].default
    ]
}]

export { fruits as default }