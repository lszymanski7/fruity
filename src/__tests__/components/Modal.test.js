import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import Modal from '../../components/Modal'
import fruits from '../__fixtures__/fruits'

const props = {
	selectedFruit: fruits[0],
    isOpenModal: true,
	closeModal: jest.fn()
}

describe('Modal', () => {
	it('should be rendered correctly.', () => {
		const component = render(<Modal {...props} />)
		expect(component).toMatchSnapshot()
	})

    it('should have the correct title.', () => {
		render(<Modal {...props} />)
		const title = props.selectedFruit.name
		const h2 = screen.getByRole('heading', { level: 2 })
		expect(h2).toHaveTextContent(title)
	})

	it('should have the correct image.', () => {
		render(<Modal {...props} />)
		const img = screen.getByRole('img')
		expect(img).toHaveAttribute('alt', props.selectedFruit.name)
		expect(img).toBeInTheDocument()
	})
})

describe('Button', () => {
    it('should handle clicking.', () => {
		render(<Modal {...props} />)
		const button = screen.getByRole('button')
		fireEvent.click(button)
		expect(props.closeModal).toHaveBeenCalled()
	})
})
