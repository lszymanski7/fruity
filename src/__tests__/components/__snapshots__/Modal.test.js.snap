// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should be rendered correctly. 1`] = `
<body
  class="theme-light"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay"
      style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75);"
    >
      <div
        aria-label="Application Modal"
        aria-modal="true"
        class="ReactModal__Content modal modal--light"
        role="dialog"
        tabindex="-1"
      >
        <img
          alt="apple"
          class="modal__image"
          src="apple.webp"
        />
        <span
          class="modal__caption"
        >
          APPLE
        </span>
      </div>
    </div>
  </div>
</body>
`;
